packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -822,21 +831,48 @@ export class MergeTree {
 		if (!segment || !isRemovedAndAcked(segment)) {
 			return segment;
 		}
+
+		if (this.cachedSlideDestination?.seq !== segment.removedSeq) {
+			this.cachedSlideDestination = {
+				segmentToSlideDestination: new Map<ISegment, ISegment>(),
+				seq: segment.removedSeq,
+			};
+		}
+
+		assert(
+			this.cachedSlideDestination !== undefined,
+			"expected cachedSlideDestination to exist",
+		);
+		const cachedSegment = this.cachedSlideDestination.segmentToSlideDestination.get(segment);
```

Comment:
can you add some comments here. i think i understand how its working (you are pre-caching segments), but is not totally intuitive.

Reply:
i also wonder if you could simplify this with a bit of refactoring. in markRangeRemoved we have the full list of segments we will slide upfront, this could maybe help keep the cache local, rather than members , the other cases are a bit trickier, but overall i'd rather us not need use members we could get away with params, or block scoped variable.

Reply:
i'm pretty sure we can. All of the calls to `slideAckedRemovedSegmentReferences` invoke it repeatedly for a list of segments. Just making it take in an iterable of segments should do. There are some things to settle with iteration order (see my comment pointing out I think we have a bug when undo-ing removal of a range of segments which is concurrently removed).

If we track more information about which segment iterables are contiguous we could probably even eliminate the cache. but that's a fair bit more involved.

I think doing this in a follow-up could be reasonable. don't feel like the current diff is too awful.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/4030 to track if we check in in current state